### DISCUSSION TITLE
Starting minimized

### DISCUSSION THREAD

**Discussion by PH5HP** [rejetto/hfs#530] [General]
Hi Rejetto,

I was looking for a way to start the console of HFS minimized, but I can not find a way to make this happen. Can this be done? I looked at the options of my shortcut, but I could not find an option there to do it.

Kind regards,
Henk - https://fileserver.ph5hp.nl

**Comment by rejetto** [comment]
maybe?
<img width="347" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/be6ca18f-eb35-458e-ba95-94e88f155ecf">

**Reply by PH5HP** [reply]
Hi,

It turned out that I had to drag my shortcut to my desktop screen from the startup folder before I could see all the TABS. After I did that all the TABS were available and then I dragged the shortcut back to that folder. Strange. Both times it was just a shortcut. Problem solved (I think). Thank you.

Kind regards,
Henk - https://fileserver.ph5hp.nl/
![image](https://github.com/rejetto/hfs/assets/140949730/cc7c27f0-ca21-4102-bc06-6286f2dd87f8)

**Comment by PH5HP** [comment]
Hi Rejetto,

It still does not work. When I edit the shortcut to start the console of HFS minimized then it will not start at all after a boot of my PC. I have to start it manually and then will start minimized.

Any ideas?

Kind regards,
Henk - https://fileserver.ph5hp.nl/

**Comment by rejetto** [comment]
you say "will not start" but it's to be verified if it's never executed, or executed and stopped, or it's still running and you just don't see it.
1. First verify if you see a process "hfs" running, even when you think it's not running.
2. see if anything new appears in file "debug.log".
3. specify what you do to make it run at boot.

**Reply by PH5HP** [reply]
Hi,

1. No HFS process is running.
2. Nothing new related to this in the debug.log.
3. I have a shortcut (HFS.LNK) in the STARTUP folder of Windows 10 64-bit.

About 3:
The STARTUP folder is the folder you get if you goto WINDOWS KEY + R -> shell:startup. In that folder I have the HFS.LNK file. In properties of that file I have it set to start minimized. If I double click on it to start HFS it starts minimized without any problem. Strangly it does not start automaticly when I reboot the computer.

Kind regards,
Henk - https://fileserver.ph5hp.nl/

**Comment by rejetto** [comment]
no idea.
if there was an error it would be visible in debug.log
did you consider making it a service?

**Reply by PH5HP** [reply]
Hi,

Minimized or not is not a big deal. :-)

In the past I did try to make it a service I think. At the moment I do not know how this is done anymore.

Kind regards,
Henk - https://fileserver.ph5hp.nl/

**Reply by PH5HP** [reply]
Hi,

I am trying to install HFS as a service, but I have still no succes.

I tried:
sc.exe create HFS binpath= "C:\Communication\HFS\new-hfs.exe"

But this does not work.

Kind regards,
Henk - https://fileserver.ph5hp.nl/

**Reply by TEA-Time1** [reply]
Hey Henk,

I don't know all the ins and outs because I've never needed to make an executable run as a service myself, but I do know that an executable has to have something special about it to run as a service. But there's a solution for "standard" executables called NSSM (Non-Sucking Service Manager (because the previous solution "srvany" must have sucked. heh)). You can find NSSM here:

https://nssm.cc/

-Tim

**Reply by PH5HP** [reply]
Hi Tim,

I used NSSM and it worked the first time. Thank you. HFS is now running as a service.

Kind regards,
Henk - https://fileserver.ph5hp.nl/

**Reply by TEA-Time1** [reply]
Cool, you're welcome.

**Comment by rejetto** [comment]
thanks for sharing, gents.
the downside of having it the exe as a service is that's harder to update.
I disabled automatic updates in most recent versions when hfs finds it is running as a service, just to avoid failure, as I undestand the way i do it is not compatible with services.
So, to update, i guess you have to stop the service, replace the exe yourself, and the start it.
I'm not sure what happens with nssm, i didn't try myself.

i have little experience with services myself, but if you manage to get this command to work `npx -y hfs@latest`
or `@beta` for the bolder ones.
this way, i understand, should grant automatic update at each restart.
surely one day i should look into the matter.

**Reply by PH5HP** [reply]
Hi,

I will let you know if we get a new version of HFS.  :-) 

Kind regards,
Henk - https://fileserver.ph5hp.nl/
